packages/x-data-grid/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -464,19 +468,19 @@ export const useGridColumnResize = (
 
     refs.columnHeaderElement = findHeaderElementFromField(
       apiRef.current.columnHeadersContainerRef!.current!,
-      colDef.field,
+      escapedColDefField,
     );
 
     const headerFilterElement = root.querySelector(
-      `.${gridClasses.headerFilterRow} [data-field="${colDef.field}"]`,
+      `.${gridClasses.headerFilterRow} [data-field="${escapedColDefField}"]`,
     );
     if (headerFilterElement) {
       refs.headerFilterElement = headerFilterElement as HTMLDivElement;
     }
 
     refs.groupHeaderElements = findGroupHeaderElementsFromField(
       apiRef.current.columnHeadersContainerRef?.current!,
-      colDef.field,
+      escapedColDefField,
```

Comment:
It feels like the escaping should happen in `findGroupHeaderElementsFromField()`. Otherwise, we will reintroduce the bug the next time the helper is used.